### PR TITLE
Feature/offer market

### LINKED ISSUE #1: Support purchase offer data
## Context

 We need to indexing the data from purchase offer contract to support the Offer Market features,

## Related Events:

```
event PurchaseOfferCancelled(address indexed creator, uint256 offerId, uint256 penalty);
event PurchaseOfferCreated(
    address consumer,
    bytes32 deploymentId,
    uint256 planTemplateId,
    uint256 deposit,
    uint16 limit,
    uint256 minimumAcceptHeight,
    uint256 expireDate
);
```

### PR DESCRIPTION
**Pull request by heyacherry**

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -163,3 +158,25 @@ type IndexerReward @entity {
...
+  minimumAcceptHeight: BigInt!
+  expireDate: Date!
+  limit: Int! # indexer cap
+  accepted: Int! # accepted indexer amount
+  reachLimit: Boolean! # whether reach limit
+  withdrawn: Boolean! # withdraw by cancel event
+  withdrawPenalty: BigInt
+}
+
+type OfferAccepted @entity {
```
Comment (mzxyz) [inline schema.graphql]:
```suggestion
type AcceptedOffer @entity {
```

File: schema.graphql [hunk 2]
```diff
@@ -163,3 +158,25 @@ type IndexerReward @entity {
   additions: BigInt!
   removals: BigInt!
 }
+
+type Offer @entity {
+  id: ID! # offerID
+  consumer: String!
+  deployment: Deployment!
+  planTemplate: PlanTemplate!
```
Comment (mzxyz) [inline schema.graphql:166]: Is this mean to be `planTemplateId`?

File: src/mappings/purchaseOffer.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export async function handlePurchaseOfferCancelled(
+  event: AcalaEvmEvent<PurchaseOfferCancelledEvent['args']>
+): Promise<void> {
+  logger.info('handlePurchaseOfferCancelled');
+  assert(event.args, 'No event args');
+
+  const offer = await Offer.get(event.args.offerId.toString());
+  assert(offer, `offer not found. planId="${event.args.offerId.toString()}"`);
```
Comment (mzxyz) [inline src/mappings/purchaseOffer.ts]:
```suggestion
  assert(offer, `offer not found. offerId="${event.args.offerId.toString()}"`);
```

File: schema.graphql (hunk 2 above)
Comment (heyacherry) [reply schema.graphql:166]: No. It is supposed to connect `PlanTemplate` entity.

File: schema.graphql (hunk 1 above)
Comment (heyacherry) [reply schema.graphql]: Updated! Thanks.

File: src/mappings/purchaseOffer.ts (hunk 3 above)
Comment (heyacherry) [reply src/mappings/purchaseOffer.ts]: Updated! Thanks.

**Review by mzxyz — APPROVED** [review]

**Pull request merged**